ethcore/src/engines/clique/tests.rs

```diff
@@ -0,0 +1,753 @@
+	pub signers: HashMap<char, KeyPair>,
+}
+
+impl Tester {
+	fn with(epoch: u64, period: u64, secrets: Vec<(char, bool)>) -> Self {
+		let clique = Clique::with_test(epoch, period);
+		let mut genesis = Header::default();
+		let mut signers = HashMap::new();
+
+		let call = |_a, _b| {
+			unimplemented!("Clique doesn't use Engine::Call");
+		};
+
+		let mut extra_data = vec![0; VANITY_LENGTH];
+
+		for (id, in_signer_list) in secrets.iter() {
+			let secret = Secret::from(H256::from(*id as u64));
+			let keypair = KeyPair::from_secret(secret).unwrap();
+			if *in_signer_list {
+				extra_data.extend(&*keypair.address());
+			}
+			signers.insert(*id, keypair);
+		}
+
+		// append dummy signature
+		extra_data.extend(std::iter::repeat(0).take(SIGNATURE_LENGTH));
+
+		genesis.set_extra_data(extra_data);
+		genesis.set_gas_limit(U256::from(0xa00000));
+		genesis.set_difficulty(U256::from(1));
+		genesis.set_seal(util::null_seal());
+
+		clique.genesis_epoch_data(&genesis, &call).expect("Create genesis failed");
+		Self {clique, genesis, db: get_temp_state_db(), signers}
+	}
+}
+
+impl Tester {
+	fn get_difficulty(&self, block_num: BlockNumber, header: &Header, signer: &Address) -> U256 {
+		let state = self.clique.state(header).unwrap();
+		if state.is_inturn(block_num, signer) {
+			DIFF_INTURN
+		} else {
+			DIFF_NOTURN
+		}
+	}
+
+	// Note, this will read the cache and `will` not work with more than 128 blocks
+	fn get_state_at_block(&self, hash: &H256) -> CliqueBlockState {
+		self.clique.block_state_by_hash.write()
+			.get_mut(hash)
+			.expect("CliqueBlockState not found tested failed")
+			.clone()
+	}
+
+	// Get signers after a certain state
+	// This is generally used to fetch the state after a test has been executed and checked against
+	// the intial list of signers provided in the test
+	fn clique_signers(&self, hash: &H256) -> Vec<Address> {
+		self.get_state_at_block(hash).signers().to_vec()
+	}
+
+	// Fetches all addresses at current `block` and converts them back to `tags (char)` and sorts them
+	// Addresses are supposed sorted based on address but these tests are using `tags` just for simplicity
+	// and the order is not important!
+	fn tags_from_vec(&self, addr: &Vec<Address>) -> Vec<char> {
+		let mut tags: Vec<char> = addr.iter().filter_map(|addr| {
+			for (t, kp) in self.signers.iter() {
+				if addr == &kp.address() {
+					return Some(*t)
+				}
+			}
+			None
+		}).collect();
+		tags.sort();
+		tags
+	}
+
+	fn new_block_and_import(
+		&self,
+		block_type: CliqueBlockType,
+		last_header: &Header,
+		beneficary: Option<Address>,
+		signer: char,
+		) -> Result<OpenBlock, Error> {
+
+		let mut extra_data = vec![0; VANITY_LENGTH];
+		let mut seal = util::null_seal();
+		let last_hash = last_header.hash();
+
+		match block_type {
+			CliqueBlockType::Checkpoint => {
+				let signers = self.clique.state(&last_header).unwrap().signers().clone();
+				for signer in signers {
+					extra_data.extend(&*signer);
+				}
+				{}
```

Comment:
What's up with this?

Reply:
it was just a needless block to return the `unit type` fixed now!